src/zenml/integrations/kubernetes/orchestrators/kube_utils.py

```diff
@@ -450,6 +450,39 @@ def create_or_update_secret(
         update_secret(core_api, namespace, secret_name, data)
 
 
+def create_or_update_secret_from_manifest(
```

Comment:
There already is a `create_secret` function in this file, can you either use that or adjust it so we have less duplicated code?